### PR TITLE
Use freeze end tick instead of a flag. Fix #1659

### LINKED ISSUE #1659: Character flags
Freeze should maybe be a "ticks remaining" or "end tick" instead of a flag (for proper prediction).

Spectating should not be a character flag, should be a player flag (as it has nothing to do with the physical character).

### PR DESCRIPTION
**Pull request by Learath2**
I used -1 for deep frozen, and 0 for not frozen. Not really sure if this is the best way.

### CODE REVIEW

File: src/game/gamecore.h [hunk 1]
```diff
@@ -246,7 +246,7 @@ class CCharacterCore
 	bool m_HasTelegunGun;
 	bool m_HasTelegunGrenade;
 	bool m_HasTelegunLaser;
-	bool m_Frozen;
+	bool m_FreezeEnd;
```
Comment (Ryozuki) [inline src/game/gamecore.h]: shouldn't this be a int?

File: src/game/server/entities/character.cpp [hunk 2]
```diff
@@ -1187,25 +1187,21 @@ void CCharacter::Snap(int SnappingClient)
 	if(m_Jetpack || m_NinjaJetpack)
 		pDDNetCharacter->m_Flags |= CHARACTERFLAG_JETPACK;
 	if(m_FreezeTime > 0)
-		pDDNetCharacter->m_Flags |= CHARACTERFLAG_FROZEN;
-	if(m_DeepFreeze)
-		pDDNetCharacter->m_Flags |= CHARACTERFLAG_DEEP_FROZEN;
-	if(m_Hit&DISABLE_HIT_GRENADE)
```
Comment (heinrich5991) [inline src/game/server/entities/character.cpp]: Wrong `if` deleted.

**Comment by Learath2** [comment]
Well `m_DeepFreeze` is not new. I did want to revamp freeze code, but I'm not sure if that's a great idea, it might change some edge case behaviour (e.g. deepfly).

**Comment by heinrich5991** [comment]
Would `bors r+` but it's a draft PR.

**Comment by Learath2** [comment]
@heinrich5991 r+ if you think using `-1` as DF is good.

**Pull request merged**